pkg/apis/pipeline/v1beta1/when_types.go

```diff
@@ -158,3 +158,13 @@ func (wes WhenExpressions) ReplaceWhenExpressionsVariables(replacements map[stri
 	}
 	return replaced
 }
+
+// WhenSkippedPolicy describes what should happen when a Task is skipped
+// because its WhenExpressions evaluated to false. WhenSkippedPolicy can
+// be specified only in Tasks that are guarded with WhenExpressions and do not
+// have resource dependencies.
+type WhenSkippedPolicy string
+
+const (
+	RunBranch WhenSkippedPolicy = "runBranch"
```

Comment:
Thanks for the review @sbwsg! 

I'd been conflicted about whether to include `skipBranch` or not, given it's the default behavior in all other cases, so added `runBranch` only to keep it simple because that's the only new behavior we're adding -- though the example you've provided makes me lean towards adding the `skipBranch` as well

Will capitalize the aliases to `RunBranch` and `SkipBranch`

Reply:
yup I agree with @sbwsg we need a default state set to `SkipBranch` but I am leaning towards not adding default as well  🙃  

defaults makes more sense in case of `boolean` but with string types, if its not needed, we can avoid setting defaults.

`default` will make code even more complicated. We can may be change the `TEP` to not include `skipBranch`, thoughts?

Reply:
when `when` expression evaluates to false, by default, the entire branch is skipped, if we want to enable it, use `runBranch` policy in `whenSkipped`